synapse/federation/transport/client.py

```diff
@@ -47,7 +47,7 @@ class TransportLayerClient:
 
     def __init__(self, hs):
         self.server_name = hs.hostname
-        self.client = hs.get_federation_http_client()
+        self.client: MatrixFederationHttpClient = hs.get_federation_http_client()
```

Comment:
it'd be better to annotate `hs` to be a `HomeServer`, then this would magically work.